bee-block/src/protocol.rs

```diff
@@ -27,3 +27,25 @@ pub struct ProtocolParameters {
     // TokenSupply defines the current token supply on the network.
     token_supply: u64,
 }
+
+#[cfg(feature = "inx")]
+impl TryFrom<inx::proto::ProtocolParameters> for ProtocolParameters {
+    type Error = crate::error::inx::InxError;
+
+    fn try_from(value: inx::proto::ProtocolParameters) -> Result<Self, Self::Error> {
+        Ok(Self {
+            version: value.version as u8,
+            network_name: <StringPrefix<u8>>::try_from(value.network_name)
+                .map_err(|e| Self::Error::InvalidString(e.to_string()))?,
+            bech32_hrp: <StringPrefix<u8>>::try_from(value.bech32_hrp)
+                .map_err(|e| Self::Error::InvalidString(e.to_string()))?,
+            min_pow_score: value.min_po_w_score,
+            below_max_depth: value.below_max_depth as u8,
+            rent_structure: value
+                .rent_structure
+                .ok_or(Self::Error::MissingField("rent_structure"))?
```

Comment:
Why is this an option?

Reply:
Because everything that's not a primitive type in protobuf is an `Option`. @pvdrz has some lovely things to say about that :trollface: